components/Habit.js

```diff
@@ -8,15 +8,28 @@ export default function Habit(props) {
   const [checkmark, setCheckmark] = useState('/grayCheckMark.svg');
   const [isEmojiModalShowing, setIsEmojiModalShowing] = useState(false);
   const [emoji, setEmoji] = useState(props.habit.emoji);
+  const [textDecoration, setTextDecoration] = useState('none');
 
-  const deleteHandler = () => props.deleteHandler(props.id);
-
-  const styleCheckmark = () => {
+  const deleteHandler = (e) => {
+    e.stopPropagation()
+    props.deleteHandler(props.id);
+  }
+  const completeHabit = () => {
     if (checkmark === "/redCheckMark.svg") {
       setCheckmark("/grayCheckMark.svg");
     } else {
       setCheckmark("/redCheckMark.svg");
     }
+    if (props.habit.completed === false) {
+      props.updateCompletedHabits(props.id, true);
+    } else {
+      props.updateCompletedHabits(props.id, false);
+    }
+    if (textDecoration === 'none') {
+      setTextDecoration('line-through');
+    } else {
+      setTextDecoration('none')
+    }
```

Comment:
This whole function would be a lot easier to read if the logic was grouped based on whether we are setting things to true or false, example:

```
const completeHabit = () => {
    if (props.habit.completed === false) {
      props.updateCompletedHabits(props.id, true);
      setCheckmark("/grayCheckMark.svg");
      setTextDecoration('line-through');
    } else {
     ....
    }
   ```

Doing it this way lets you 1) reduce the number of if/elses 2) Makes it easier to see what's happening in when a user does some action.

Reply:
Done